src/main/scala/is/hail/expr/package.scala

```diff
@@ -21,6 +22,18 @@ package object expr extends HailRepFunctions {
     def copy(): TypedAggregator[S]
   }
 
+  trait RegionValueAggregator extends Serializable {
+    def typ: Type
```

Comment:
I don't think this should have the type.  RVA should be a (Hail) runtime object, but `Type` is a (Hail) compile-time object.

Reply:
This is reasonable. I don't need the type at Hail runtime, I was piping pairs of type and RVA around a fair bit so I moved the type onto the RVA for ease of programming and reasoning about correctness. I will divorce them again so as to avoid serialization of the Type and to avoid mixing of concerns.

Reply:
> I was piping pairs of type and RVA around a fair bit

I'm still confused by this.  One is a (Hail) compile time object, and one is a runtime object.  How are they in the same tuple?

Reply:
In `ExtractAggregators`, I combine the output of all the aggregations into a single Hail struct that is passed as the input to (what I'm calling) the stage0 IR (i.e. the fragment of the IR that consume aggregated results and produces the final value). I need to know each aggregator's type so I can create a `TStruct` with a field for each aggregation so that I can then reference fields of that aggregation in the stage0 IR. The types of the region value aggregators are not needed after `ExtractAggregators.apply` exits. I can remove it from RVA and pass tuples around until that point and after that point only have RVA.

Reply:
I realized I already had another case class that was carrying some other compile-time metadata, so I added it to that.

Reply:
Ok this should be addressed now